templates/config/routes.rb

```diff
@@ -1,15 +1,15 @@
 # frozen_string_literal: true
 
-Spree::Core::Engine.routes.draw do
+Rails.application.routes.draw do
   root to: 'home#index'
 
   devise_for(:spree_user, {
     class_name: 'Spree::User',
     controllers: {
-      sessions: 'spree/user_sessions',
-      registrations: 'spree/user_registrations',
-      passwords: 'spree/user_passwords',
-      confirmations: 'spree/user_confirmations'
+      sessions: 'user_sessions',
+      registrations: 'user_registrations',
+      passwords: 'user_passwords',
+      confirmations: 'user_confirmations'
```

Comment:
Currently, this config produces Devise routes that still have `spree_user` in the URLs:

```
                                  Prefix Verb   URI Pattern                                                                                       Controller#Action
                                    root GET    /                                                                                                 home#index
                  new_spree_user_session GET    /user/spree_user/sign_in(.:format)                                                                user_sessions#new
                      spree_user_session POST   /user/spree_user/sign_in(.:format)                                                                user_sessions#create
              destroy_spree_user_session DELETE /user/spree_user/logout(.:format)                                                                 user_sessions#destroy
                 new_spree_user_password GET    /user/spree_user/password/new(.:format)                                                           user_passwords#new
                edit_spree_user_password GET    /user/spree_user/password/edit(.:format)                                                          user_passwords#edit
                     spree_user_password PATCH  /user/spree_user/password(.:format)                                                               user_passwords#update
                                         PUT    /user/spree_user/password(.:format)                                                               user_passwords#update
                                         POST   /user/spree_user/password(.:format)                                                               user_passwords#create
          cancel_spree_user_registration GET    /user/spree_user/cancel(.:format)                                                                 user_registrations#cancel
             new_spree_user_registration GET    /user/spree_user/sign_up(.:format)                                                                user_registrations#new
            edit_spree_user_registration GET    /user/spree_user/edit(.:format)                                                                   user_registrations#edit
                 spree_user_registration PATCH  /user/spree_user(.:format)                                                                        user_registrations#update
                                         PUT    /user/spree_user(.:format)                                                                        user_registrations#update
                                         DELETE /user/spree_user(.:format)                                                                        user_registrations#destroy
                                         POST   /user/spree_user(.:format)                                                                        user_registrations#create
```

I tried changing the config to

```
  devise_for(:user, {
    class_name: 'Spree::User',
    controllers: {
      sessions: 'user_sessions',
      registrations: 'user_registrations',
      passwords: 'user_passwords',
      confirmations: 'user_confirmations'
    },
    skip: [:unlocks, :omniauth_callbacks],
    path_names: { sign_out: 'logout' }
  })
```

However, this causes the specs to fail. It seems to break `spree_current_user`. I still need to investigate this.

Considering that the frontend ignores these Devise URLs in favor of custom URLs (`get '/login'`), I think it's tolerable at the moment to keep the `spree_user` in these Devise routes.

Let me know what you think!

Reply:
I think this may be solved with the `path` options: https://www.rubydoc.info/github/heartcombo/devise/main/ActionDispatch/Routing/Mapper%3Adevise_for. Let me know!

Reply:
I'll give it one more try! TODO